editor/filesystem_dock.cpp

```diff
@@ -1472,12 +1472,22 @@ void FileSystemDock::_folder_removed(String p_folder) {
 
 void FileSystemDock::_rename_operation_confirm() {
 	String new_name = rename_dialog_text->get_text().strip_edges();
+	String old_name = tree->get_selected()->get_text(0);
 	if (new_name.length() == 0) {
 		EditorNode::get_singleton()->show_warning(TTR("No name provided."));
 		return;
 	} else if (new_name.find("/") != -1 || new_name.find("\\") != -1 || new_name.find(":") != -1) {
 		EditorNode::get_singleton()->show_warning(TTR("Name contains invalid characters."));
 		return;
+	} else if (old_name.get_extension() != new_name.get_extension()) {
+		List<String> extensions;
+		ResourceLoader::get_recognized_extensions_for_type("", &extensions);
+		const Vector<String> textfile_ext = ((String)(EditorSettings::get_singleton()->get("docks/filesystem/textfile_extensions"))).split(",", false);
+		bool extension_unkown = !extensions.find(new_name.get_extension()) && textfile_ext.find(new_name.get_extension()) == -1;
+		if (to_rename.is_file && extension_unkown) {
```

Comment:
`to_rename.is_file` could be moved to the condition above, to avoid checking extension when it's not necessary.
Then the `extension_unkown` variable is not needed.

Reply:
Done.